### DISCUSSION TITLE
User feedback and FAQ

### DISCUSSION THREAD

**Discussion by mooch443** [mooch443/trex#85] [Announcements]
## Welcome
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members. 

We also hope to use this place to collect and resolve frequently asked questions. GitHub issues are a great start, but they usually "disappear" once an individual issue has been solved - this page, however, will stay alive beyond that point and is thus a great place for new users to look whether their question has already been asked before.

Some example issues that might be of broader interest:

#41 Cannot get auto_train working
#55 blob_split
#56 Difficulties finding good parameters
#74 Common background in batch processing

Issues can also be converted into discussions easily (if it turns out that their scope is broader than expected). Feel free to share any thoughts and questions you might have.

**Comment by walsmanjason** [comment]
When I train individual IDs on Trex, the GUI displays "CPU" in the lower right where it should display the name of my graphics card. From the documentation, I understand that this is because I am using Windows and an Intel graphics-card. I am looking into purchasing an NVIDIA graphics-card. Do you have any suggestions/warnings about what to look for in an NVIDIA graphics-card?

**Comment by mooch443** [comment]
Yes, exactly! Intel graphics cards do not "count" in this case (also they aren't extremely powerful). The type of graphics card is not hugely important, but cards with more memory let you use more samples at once for training (e.g. also for larger groups). I am currently using a 1080 TI / 3080 something. The Ti is specifically because of larger memory.

**Reply by walsmanjason** [reply]
Thank you!

We are also considering using Trex on a Mac OS as well. The documentation does not suggest a graphics card for Apple Silicone, as far as I can tell. Does that mean that it should work with most Apple Silicone graphics cards?

**Reply by mooch443** [reply]
Apple silicone has their very own implementations - if you are referring to whether M1 Max or pro are better, there is no difference except in speed. As far as I can tell. The plug-in is by apple and should support their hardware. I have to add though that I have no experience with M2s of course.

**Comment by shivChitinous** [comment]
I'm using tgrabs+trex to track some stationary and some moving individuals. All stationary individuals make it into the background image and don't get segmented out, was wondering if there is a way to modify the background image so that small regions can be "cleaned up" by using the average brightness of the surrounding regions (as in ctrax) or if there is a good way to make sure that static individuals with high contrast don't make it into the background. Separately, is there a way to apply a custom mask so that only individuals within that region get segmented out?

**Reply by mooch443** [reply]
Hey, sorry for the late reply. In case this is still relevant: you can use ``max`` (for dark) or ``min`` threshold (for light backgrounds), which should avoid high contrast individuals to become the background - that is, if they move at all. Otherwise you could also disable the difference calculation using ``enable_difference = false``. There is also an adaptive threshold implementation, although I hadn't tested it in a while. In TGrabs enable it by setting ``use_adaptive_threshold = true``, for which you can set a neighborhood size setting ``adaptive_threshold_scale`` to a percentage of the background image size (0.1 by default).

Regarding custom mask: You can have a video mask (``mask_path``) in TGrabs, and theoretically you can have static masks... but currently no implementation to make them :S It's part of the video format, but hasnt been implemented. As a replacement, though, it's easy to use custom shapes during tracking instead: [docs](https://trex.run/docs/gui.html#excluding-regions-tracking-specific-regions-exclusively).
